test/test_alert_manager.cpp

```diff
@@ -223,6 +255,42 @@ TORRENT_TEST(extensions)
 	TEST_EQUAL(plugin_alerts[0], 70);
 	TEST_EQUAL(plugin_alerts[1], 70);
 	TEST_EQUAL(plugin_alerts[2], 70);
+
+	for (int i = 0; i < 35; ++i)
+		mgr.emplace_alert<torrent_finished_alert>(torrent_handle());
+
+	TEST_EQUAL(plugin_alerts[0], 100);
+	TEST_EQUAL(plugin_alerts[1], 100);
+	TEST_EQUAL(plugin_alerts[2], 105);
+
+	mgr.set_alert_queue_size_limit(0);
+
+	for (int i = 0; i < 35; ++i)
+		mgr.emplace_alert<torrent_finished_alert>(torrent_handle());
+
+	TEST_EQUAL(plugin_alerts[0], 100);
+	TEST_EQUAL(plugin_alerts[1], 100);
+	TEST_EQUAL(plugin_alerts[2], 140);
+
+	bool running = false;
+	int num_resume = 0;
+	std::vector<alert*> alerts;
+	mgr.get_all(alerts, num_resume);
+	libtorrent::thread t(boost::bind(&alert_popper, boost::ref(mgr), boost::ref(running)));
+
+	/* make sure the thread is started */
+	while (!running) test_sleep(10);
+
+	for (int i = 0; i < 100000000; ++i)
```

Comment:
since this literal is greater than what ``int`` can represent, it may yield warnings, or possibly errors. Do you really need this many iterations?